### DISCUSSION TITLE
why the user proxy initial message is tagged as role "assistant" ?

### DISCUSSION THREAD

**Discussion by elexira** [microsoft/autogen#2008] [Q&A]
When i invoke a conversation 
`user_proxy.initiate_chat(assistant, message=question) `
where `user_proxy` is an instance of `UserProxyAgent`
and `assistant` is an instance of `AssistantAgent`
the very first message in the `chat_history` , which is seen as `user_proxy (to assistant):` is the message that the `user_proxy` sends to `assistant`. I was expecting it would be labeled as a `user` message but it is tagged as `{role: assistant }` . Does `UserProxyAgent`  not represent the user ? hence the very first message should be tagged as 'user`. I am confused.

**Comment by rickyloynd-microsoft** [comment]
This is a common source of confusion. When two agents are conversing, each maintains its own copy of the message history, and their roles are reversed in those two copies. The reason for this is because openai defined a fixed set of message roles like user and assistant, and autogen follows those roles in its message lists. In a group chat it may be that none of the agents represent the user, and yet every message list sent to the LLM must be annotated with user and assistant roles. The simple rule to remember is that every agent carries the assistant role in its own copy of the message list.

**Reply by elexira** [reply]
chat history is not a method for assistant agent only for the user proxy agent ....
here is a bigger issue, that is 

imagine there is a conversation history up to now as follows 

  messages=[
    {"role": "system", "content": "Given the history of chat and user question what is the user asking ?"},
    {"role": "user", "content": "Who won the world series in 2020?"},
    {"role": "assistant", "content": "The Los Angeles Dodgers won the World Series in 2020."},
    {"role": "user", "content": "Where was it played?"}
  ]
)

and i want to provide this as a seed to user_proxy.initiate_chat(assistant, messages)

where messages under role `user` are from the user_proxy agent and the ones from the 'assistant` are from the assistant agent. Can i achieve this ? i cannot handle this with history because new agents are creates every-time there is a call for generating response.
